apps/internal/json/types/time/time.go

```diff
@@ -59,12 +60,48 @@ func (d DurationTime) MarshalJSON() ([]byte, error) {
 	return []byte(fmt.Sprintf("%d", int64(dt*time.Second))), nil
 }
 
-// UnmarshalJSON implements encoding/json.UnmarshalJSON().
-func (d *DurationTime) UnmarshalJSON(b []byte) error {
-	i, err := strconv.Atoi(strings.Trim(string(b), `"`))
+// UnmarshalJSON custom unmarshaler for DurationTime
+func (t *DurationTime) UnmarshalJSON(b []byte) error {
+	// Remove the quotes around the JSON string
+	str := strings.Trim(string(b), `"`)
+
+	// Try parsing as Unix timestamp (seconds since the Unix epoch)
+	if len(str) == 10 {
+		if unixTimestamp, err := strconv.ParseInt(str, 10, 64); err == nil {
+			t.T = time.Unix(unixTimestamp, 0)
+			return nil
+		}
+	}
+
+	// Try parsing as ISO 8601 format (e.g., "2024-10-18T19:51:37.0000000+00:00")
+	iso8601Layout := "2006-01-02T15:04:05.9999999-07:00"
```

Comment:
Why do we need to parse these other formats?

Reply:
Where are these other formats coming from? I haven't seen anything like them in a token response, excepting those from an old version of App Service we don't support